Taarafo.Core.Tests.Unit/Services/Foundations/Comments/CommentServiceTests.Exceptions.Add.cs

```diff
@@ -194,5 +194,50 @@ public async Task ShouldThrowServiceExceptionOnAddIfDatabaseUpdateErrorOccursAnd
             this.storageBrokerMock.VerifyNoOtherCalls();
             this.loggingBrokerMock.VerifyNoOtherCalls();
         }
+
+        [Fact]
+        public async void ShouldThrowValidationExceptionOnAddIfReferenceErrorOccursAndLogItAsync()
+        {
+            // given
+            DateTimeOffset randomDateTime =
+                GetRandomDateTimeOffset();
+
+            Comment randomComment = CreateRandomComment(randomDateTime);
+            Comment someComment = randomComment;
+            string randomMessage = GetRandomMessage();
+            string exceptionMessage = randomMessage;
+            var foreignKeyConstraintConflictException =
+                new ForeignKeyConstraintConflictException(exceptionMessage);
+
+            var invalidCommentReferenceException =
+                new InvalidCommentReferenceException(foreignKeyConstraintConflictException);
+
+            var expectedCommentValidationException =
+                new CommentDependencyValidationException(invalidCommentReferenceException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                .Throws(foreignKeyConstraintConflictException);
```

Comment:
You need a tab here. Do the same for the rest please.

Reply:
@hassanhabib done